### PR TITLE
Learning curve feature

### PR DESCRIPTION
**Pull request by lhjohn**
This pull request contains a limited number of commits that summarize the entire work on learning curves done within the Erasmus MC fork.

As a side note: although Travis could not find any problem, I was unable to successfully run 'Test package' and 'Check package' in RStudio. In an effort to find the problem I also made some minor modifications to the testthat tests, by putting the context() first, as suggested by Hadley Wickham (https://github.com/r-lib/testthat/issues/700).

### CODE REVIEW

File: R/DataSplitting.R [hunk 1]
```diff
@@ -166,21 +194,28 @@ timeSplitter <- function(population, test = 0.3, nfold = 3, seed = NULL) {
 
   nonPpl.group <- rep(-1, nrow(nonPpl))
   nonPpl.group[nonPpl$date <= testDate] <- rep(1:nfold,
-                                               each = ceiling(sum(nonPpl$date <= testDate)/nfold))[1:sum(nonPpl$date <=
+                                               each = ceiling(sum(nonPpl$date <= testDate)*(train/(1-test))/nfold))[1:sum(nonPpl$date <=
```
Comment (PRijnbeek) [inline R/DataSplitting.R]: Add subsampling randomly

File: R/EvaluatePlp.R [hunk 2]
```diff
@@ -615,9 +629,8 @@ getDemographicSummary <- function(prediction, plpData){
       
       ageCovariates <- plpData$covariates[ffbase::`%in%`(plpData$covariates$covariateId, 
                                                          plpData$covariateRef$covariateId[plpData$covariateRef$analysisId==3]), ]
-      #ageCovariates <- ageCovariates[ffbase::`%in%`(ageCovariates$rowId, prediction$rowId), ]
-      #ageCovariates <- ff::as.ram(ageCovariates)
-      ageCovariates <- ff::as.ram(ageCovariates)[ff::as.ram(ageCovariates$rowId%in%prediction$rowId),]
```
Comment (PRijnbeek) [inline R/EvaluatePlp.R]: Keep line 620

**Comment by lhjohn** [comment]
Hi @jreps,
we have finalized and tested the code base for the learning curve feature. Could you go ahead and review the pull request? Let us know, if there are modifications necessary before merging it into the master.

**Comment by lhjohn** [comment]
Will be reopened with additional commits from the Erasmus MC fork.

**Pull request closed without merging**